x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateTrunc.java

```diff
@@ -71,17 +78,13 @@ static long process(long fieldVal, @Fixed Rounding.Prepared rounding) {
     }
 
     @Override
-    protected BinaryScalarFunction replaceChildren(Expression newLeft, Expression newRight) {
-        return new DateTrunc(source(), newLeft, newRight);
+    public Expression replaceChildren(List<Expression> newChildren) {
+        return new DateTrunc(source(), newChildren.get(0), newChildren.get(1));
     }
 
     @Override
     protected NodeInfo<? extends Expression> info() {
-        return NodeInfo.create(this, DateTrunc::new, interval(), timestampField());
-    }
-
-    public Expression interval() {
-        return left();
+        return NodeInfo.create(this, DateTrunc::new, children().get(0), children().get(1));
```

Comment:
This is incorrect, an interval needs to be returned, not a new DateTrunc.

Reply:
The interval() method is deleted.

Reply:
My bad!